src/test/java/org/example/UserTest.java

```diff
@@ -16,45 +16,94 @@
 
     }
 
     @Test
-    public void getUser() throws ClientProtocolException, IOException {
-        HttpUriRequest request = new HttpGet("https://gorest.co.in/public/v2/users/7824865");
+    public void getUser() throws IOException {
+        HttpUriRequest request = new HttpGet("https://gorest.co.in/public/v2/users/7881661");
         HttpResponse httpResponse = HttpClientBuilder.create().build().execute(request);
         Assert.assertEquals(httpResponse.getCode(), 200);
     }
 
-    @Test void postUser() throws IOException {
+    @Test
+    public void getNonExistentUser() throws IOException {
+        HttpUriRequest request = new HttpGet("https://gorest.co.in/public/v2/users/1");
+        HttpResponse httpResponse = HttpClientBuilder.create().build().execute(request);
+        Assert.assertEquals(httpResponse.getCode(), 404);
+    }
+
+    @Test
+    void postUser() throws IOException {
         CloseableHttpClient httpclient = HttpClients.createDefault();
         HttpPost httpPost = new HttpPost("https://gorest.co.in/public/v2/users?access-token=5d034ab1ba63642fd87b6ff1133eb5fbadad18da23c7c44bbe1f857e25b733b7");
         List<NameValuePair> params = new ArrayList<NameValuePair>(2);
-        params.add(new BasicNameValuePair("id", "12345"));
         params.add(new BasicNameValuePair("name", "John Doe"));
-        params.add(new BasicNameValuePair("email", "mail@gmail.com"));
+        params.add(new BasicNameValuePair("email", "maisli@gmail.com"));
         params.add(new BasicNameValuePair("gender", "male"));
         params.add(new BasicNameValuePair("status", "active"));
         httpPost.setEntity(new UrlEncodedFormEntity(params));
         HttpResponse response = httpclient.execute(httpPost);
         Assert.assertEquals(response.getCode(), 201);
     }
 
+    @Test
+    void postUserWithIncorrectParams() throws IOException {
+        CloseableHttpClient httpclient = HttpClients.createDefault();
+        HttpPost httpPost = new HttpPost("https://gorest.co.in/public/v2/users?access-token=5d034ab1ba63642fd87b6ff1133eb5fbadad18da23c7c44bbe1f857e25b733b7");
+        List<NameValuePair> params = new ArrayList<NameValuePair>(2);
+        params.add(new BasicNameValuePair("name", "John Doe"));
+        params.add(new BasicNameValuePair("mail", "maili@gmail.com"));
+        params.add(new BasicNameValuePair("gender", "male"));
+        params.add(new BasicNameValuePair("status", "active"));
+        httpPost.setEntity(new UrlEncodedFormEntity(params));
+        HttpResponse response = httpclient.execute(httpPost);
+        Assert.assertEquals(response.getCode(), 422);
+    }
+
+    @Test
+    void postUserWithIncorrectMailFormat() throws IOException {
+        CloseableHttpClient httpclient = HttpClients.createDefault();
+        HttpPost httpPost = new HttpPost("https://gorest.co.in/public/v2/users?access-token=5d034ab1ba63642fd87b6ff1133eb5fbadad18da23c7c44bbe1f857e25b733b7");
```

Comment:
You’re passing the token as a query param; use Authorization: Bearer header. I show you how to do it